java/java.source.base/src/org/netbeans/modules/java/source/indexing/JavaCustomIndexer.java

```diff
@@ -1395,15 +1395,7 @@ private static File dumpHeap(@NonNull final String path) {
     private static Pair<Object,Method> heapDumper;
 
     private static String computeJavacVersion() {
-        if (NoJavacHelper.hasNbJavac()) {
-            File nbJavac = InstalledFileLocator.getDefault().locate("modules/ext/nb-javac-impl.jar", "org.netbeans.modules.nbjavac.impl", false);
-            if (nbJavac != null) {
-                return String.valueOf(nbJavac.lastModified());
-            }
-            return "-1";
-        } else {
-            return System.getProperty("java.vm.version", "unknown");
-        }
+        return System.getProperty("java.vm.version", "unknown");
```

Comment:
This would mean complete re-index when running on a different JDK, even with nb-javac. That is probably too much. I guess I'd suggest to either simply return a fixed value (e.g. `return "nb-javac";`) regardless what the underlying version is, or check the version of nb-javac if installed, e.g.:
```
        String version = System.getProperty("java.vm.version", "unknown");

        for (ModuleInfo mi : Lookup.getDefault().lookupAll(ModuleInfo.class)) {
            if ("org.netbeans.libs.nbjavacapi".equals(mi.getCodeNameBase())) {
                version = mi.getSpecificationVersion().toString();
            }
        }

        return version;
```

(assuming it is able to find the wrapper module.)

Reply:
could we use SourceVersion.latest() there? This would make it nb-javac independent.

Reply:
nevermind, i believe your suggestion is better. However using the "java.version" property as fallback would make it compatible with the module spec version. It uses the same "17.0.1" format.